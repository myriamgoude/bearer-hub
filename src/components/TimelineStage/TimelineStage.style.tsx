import { css } from '@emotion/core'
import { breakpoints, colors } from '../../styles/variables'

export default {
  root: css`
    width: 100%;
    @media (max-width: ${breakpoints.lg}px) {
      padding-left: 16px !important;
    }
  `,
  stageHeading: css`
    display: flex;
    padding-top: 16px;
    color: #030d36;
    font-size: 1.2rem;

    @media (max-width: ${breakpoints.lg}px) {
      &:before {
        content: '';
        display: block;
        position: absolute;
        width: 16px;
        height: 2px;
        background: ${colors.yellow};
        left: 8px;
        margin-top: 8px;
      }
    }
  `,
  stageTooltip: css`
    display: inline-block;
    margin: 0 0.75rem;
  `,
  stageHint: css`
    display: block;
    margin-top: 1rem;
    font-size: 0.875rem;
    line-height: 1.3 !important;
  `
}
