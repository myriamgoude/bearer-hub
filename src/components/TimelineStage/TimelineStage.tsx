import * as React from 'react'
import { Container, Tooltip } from '../index'

import styled from '@emotion/styled'
import { colors, breakpoints } from '../../styles/variables'

import styles from './TimelineStage.style'

// TODO - we should move that StyledDiv to the style file.
// couldn't find how to do it :-)
const StyledDiv = styled.div<{ placement: string }>`
  @media (min-width: ${breakpoints.lg}px) {
    width: 50%;
    ${props =>
      props.placement === 'right' ? 'margin-left: auto; padding-left: calc(115px + 20px);' : 'padding-right: 125px'};
    position: relative;
    h3 {
      ${props => (props.placement === 'right' ? 'text-align: left' : 'justify-content: flex-end;text-align: right')};
    }
    &:before {
      content: '';
      display: block;
      position: absolute;
      top: 0;
      width: 115px;
      height: 40px;
      color: ${colors.yellow};
      ${props => (props.placement === 'right' ? 'padding-left:8px;' : 'padding-right:8px;')};
      ${props => (props.placement === 'right' ? 'left:-1px;' : 'right:-1px;')};
      ${props => (props.placement === 'right' ? 'text-align: left;' : 'text-align:right;')};
      ${props =>
        props.placement === 'right'
          ? `background: url(${require('../../images/curve-rtl.svg')}) no-repeat center center / contain;`
          : `background: url(${require('../../images/curve-ltr.svg')}) no-repeat center center / contain;`};
    }
  }
  @media (max-width: ${breakpoints.lg}px) {
    padding-left: 16px !important;
  }
`

interface ITimelineStageProps {
  heading: string
  tooltip?: string
  placement: string
  hint?: string
  children: any
}

const TimelineStage = (props: ITimelineStageProps) => (
  <Container style={{ marginBottom: 32 }}>
    <StyledDiv placement={props.placement}>
      <h3 css={styles.stageHeading}>
        {props.heading}
        {props.tooltip && (
          <span css={styles.stageTooltip}>
            <Tooltip placement="top" trigger={['hover']} content={props.tooltip} />
          </span>
        )}
      </h3>
      {props.children}
      {props.hint && <span css={styles.stageHint}>{props.hint}</span>}
    </StyledDiv>
  </Container>
)

export default TimelineStage
