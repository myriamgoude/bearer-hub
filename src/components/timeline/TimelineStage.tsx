import * as React from 'react'
import styled from '@emotion/styled'

import Container from '../Container'
import BackendElement from './BackendElement'
import UIElement from './UIElement'

const StyledContainer = styled(Container)`
  align-items: center;
  justify-content: center;
`

interface ITimelineStageProps {
  index: number
  stage: {
    timeToComplete: number
    backendElementType?: string
    backendElementName?: string
    uiElement?: {
      title: string
      codeSnippet: string
      image: {
        handle: string
        height: number
        width: number
      }
    }
  }
}

class TimelineStage extends React.Component<ITimelineStageProps> {
  rightAligned(index: number) {
    return index % 2 === 0
  }

  render() {
    return (
      <>
        <StyledContainer>
          <h4>{this.props.stage.timeToComplete} min(s)</h4>
        </StyledContainer>
        <Container>
          {this.props.stage.uiElement ? (
            <UIElement
              element={this.props.stage.uiElement}
              rightAligned={this.rightAligned(this.props.index) && !this.props.stage.backendElementType}
            />
          ) : null}
          {this.props.stage.backendElementType && this.props.stage.backendElementName ? (
            <BackendElement
              type={this.props.stage.backendElementType}
              name={this.props.stage.backendElementName}
              rightAligned={this.rightAligned(this.props.index) && !this.props.stage.uiElement}
            />
          ) : null}
        </Container>
      </>
    )
  }
}

export default TimelineStage
