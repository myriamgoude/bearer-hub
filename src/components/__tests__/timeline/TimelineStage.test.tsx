import React from 'react'
import renderer from 'react-test-renderer'

import TimelineStage from '../../timeline/TimelineStage'

jest.mock('../../../services/Explore', () => ({
  timer: jest.fn().mockReturnValue('11.01')
}))

describe('TimelineStage', () => {
  it('handles when UI element and backend function are both present', () => {
    const stage = {
      timeToComplete: 1,
      backendElementType: 'InstallBearerLibrary',
      backendElementName: 'connect-hello-world',

      uiElement: {
        title: 'Connect Hello World Component',
        codeSnippet: '<connect-hello-world />',
        stage: 1,
        index: 0,
        tooltip: 'This is a tooltip',
        helperText: 'This is some helper text',
        image: {
          handle: 'my-test-image',
          height: 50,
          width: 50
        }
      }
    }
    const tree = renderer.create(<TimelineStage stage={stage} index={2} timeElasped={3} />)
    expect(tree).toMatchSnapshot()
  })

  it('handles when backend function is not present, and has auto margin-left when index is even', () => {
    const stage = {
      timeToComplete: 2,

      uiElement: {
        title: 'Connect Hello World Component',
        codeSnippet: '<connect-hello-world />',
        tooltip: '',
        helperText: 'This is some helper text',
        stage: 2,
        index: 1,
        image: {
          handle: 'my-test-image',
          height: 50,
          width: 50
        }
      }
    }
    const tree = renderer.create(<TimelineStage stage={stage} index={0} timeElasped={2} />)
    expect(tree).toMatchSnapshot()
  })

  it('handles when UI element is not present, and has no auto margin when index is odd', () => {
    const stage = {
      timeToComplete: 2,
      backendElementType: 'InstallBearerLibrary',
      backendElementName: 'connect-hello-world'
    }
    const tree = renderer.create(<TimelineStage stage={stage} index={1} timeElasped={1} />)
    expect(tree).toMatchSnapshot()
  })
})
