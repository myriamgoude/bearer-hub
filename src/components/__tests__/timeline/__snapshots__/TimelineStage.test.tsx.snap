// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineStage handles when UI element and backend function are both present 1`] = `
Array [
  <div
    css={
      Array [
        Object {
          "map": undefined,
          "name": "6j1h23",
          "next": undefined,
          "styles": "
    margin: auto;
    width: 100%;
    max-width: 64rem;
  ",
        },
        undefined,
        Object {
          "map": undefined,
          "name": "8qpjjf",
          "next": undefined,
          "styles": "
    display: block;
  ",
        },
        false,
        false,
        false,
        Object {
          "map": undefined,
          "name": "1dpbaaj",
          "next": undefined,
          "styles": "
    padding: 0 0.09375rem;
  ",
        },
        undefined,
      ]
    }
  >
    <h4>
      1
       min(s)
    </h4>
  </div>,
  .emotion-1 {
  width: 50%;
}

.emotion-0 {
  font-family: 'Courier New',Courier,monospace;
}

<div
    css={
      Array [
        Object {
          "map": undefined,
          "name": "6j1h23",
          "next": undefined,
          "styles": "
    margin: auto;
    width: 100%;
    max-width: 64rem;
  ",
        },
        undefined,
        Object {
          "map": undefined,
          "name": "8qpjjf",
          "next": undefined,
          "styles": "
    display: block;
  ",
        },
        false,
        false,
        false,
        Object {
          "map": undefined,
          "name": "1dpbaaj",
          "next": undefined,
          "styles": "
    padding: 0 0.09375rem;
  ",
        },
        undefined,
      ]
    }
  >
    <div
      className="emotion-1"
    >
      <h4>
        Connect Hello World Component
      </h4>
      <div
        className=" graphcms-image-outer-wrapper"
        style={
          Object {
            "position": "relative",
            "zIndex": 0,
          }
        }
      >
        <div
          className=" graphcms-image-wrapper"
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
              "zIndex": 1,
            }
          }
        >
          <div
            style={
              Object {
                "paddingBottom": "100%",
                "width": "100%",
              }
            }
          />
          <img
            alt=""
            onLoad={null}
            src="https://media.graphcms.com/resize=w:20,h:20,fit:crop/blur=amount:2/compress/my-test-image"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "transitionDelay": "0.25s",
                "width": "100%",
              }
            }
            title=""
          />
          
        </div>
      </div>
      <div
        className="emotion-0"
      >
        &lt;connect-hello-world /&gt;
      </div>
    </div>
    <div
      className="emotion-1"
    >
      <h4>
        Install the Bearer Library
      </h4>
      <div
        className="emotion-0"
      >
        npm install @bearer/react
      </div>
    </div>
  </div>,
]
`;

exports[`TimelineStage handles when UI element is not present, and has no auto margin when index is odd 1`] = `
Array [
  <div
    css={
      Array [
        Object {
          "map": undefined,
          "name": "6j1h23",
          "next": undefined,
          "styles": "
    margin: auto;
    width: 100%;
    max-width: 64rem;
  ",
        },
        undefined,
        Object {
          "map": undefined,
          "name": "8qpjjf",
          "next": undefined,
          "styles": "
    display: block;
  ",
        },
        false,
        false,
        false,
        Object {
          "map": undefined,
          "name": "1dpbaaj",
          "next": undefined,
          "styles": "
    padding: 0 0.09375rem;
  ",
        },
        undefined,
      ]
    }
  >
    <h4>
      2
       min(s)
    </h4>
  </div>,
  .emotion-1 {
  width: 50%;
}

.emotion-0 {
  font-family: 'Courier New',Courier,monospace;
}

<div
    css={
      Array [
        Object {
          "map": undefined,
          "name": "6j1h23",
          "next": undefined,
          "styles": "
    margin: auto;
    width: 100%;
    max-width: 64rem;
  ",
        },
        undefined,
        Object {
          "map": undefined,
          "name": "8qpjjf",
          "next": undefined,
          "styles": "
    display: block;
  ",
        },
        false,
        false,
        false,
        Object {
          "map": undefined,
          "name": "1dpbaaj",
          "next": undefined,
          "styles": "
    padding: 0 0.09375rem;
  ",
        },
        undefined,
      ]
    }
  >
    <div
      className="emotion-1"
    >
      <h4>
        Install the Bearer Library
      </h4>
      <div
        className="emotion-0"
      >
        npm install @bearer/react
      </div>
    </div>
  </div>,
]
`;

exports[`TimelineStage handles when backend function is not present, and has auto margin-left when index is even 1`] = `
Array [
  <div
    css={
      Array [
        Object {
          "map": undefined,
          "name": "6j1h23",
          "next": undefined,
          "styles": "
    margin: auto;
    width: 100%;
    max-width: 64rem;
  ",
        },
        undefined,
        Object {
          "map": undefined,
          "name": "8qpjjf",
          "next": undefined,
          "styles": "
    display: block;
  ",
        },
        false,
        false,
        false,
        Object {
          "map": undefined,
          "name": "1dpbaaj",
          "next": undefined,
          "styles": "
    padding: 0 0.09375rem;
  ",
        },
        undefined,
      ]
    }
  >
    <h4>
      2
       min(s)
    </h4>
  </div>,
  .emotion-0 {
  font-family: 'Courier New',Courier,monospace;
}

.emotion-1 {
  width: 50%;
  margin-left: auto;
}

<div
    css={
      Array [
        Object {
          "map": undefined,
          "name": "6j1h23",
          "next": undefined,
          "styles": "
    margin: auto;
    width: 100%;
    max-width: 64rem;
  ",
        },
        undefined,
        Object {
          "map": undefined,
          "name": "8qpjjf",
          "next": undefined,
          "styles": "
    display: block;
  ",
        },
        false,
        false,
        false,
        Object {
          "map": undefined,
          "name": "1dpbaaj",
          "next": undefined,
          "styles": "
    padding: 0 0.09375rem;
  ",
        },
        undefined,
      ]
    }
  >
    <div
      className="emotion-1"
    >
      <h4>
        Connect Hello World Component
      </h4>
      <div
        className=" graphcms-image-outer-wrapper"
        style={
          Object {
            "position": "relative",
            "zIndex": 0,
          }
        }
      >
        <div
          className=" graphcms-image-wrapper"
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
              "zIndex": 1,
            }
          }
        >
          <div
            style={
              Object {
                "paddingBottom": "100%",
                "width": "100%",
              }
            }
          />
          <img
            alt=""
            onLoad={null}
            src="https://media.graphcms.com/resize=w:20,h:20,fit:crop/blur=amount:2/compress/my-test-image"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "transitionDelay": "0.25s",
                "width": "100%",
              }
            }
            title=""
          />
          
        </div>
      </div>
      <div
        className="emotion-0"
      >
        &lt;connect-hello-world /&gt;
      </div>
    </div>
  </div>,
]
`;
